cpp/include/cudf/sorting.hpp

```diff
@@ -390,5 +390,78 @@ std::unique_ptr<column> top_k_order(
   rmm::cuda_stream_view stream      = cudf::get_default_stream(),
   rmm::device_async_resource_ref mr = cudf::get_current_device_resource_ref());
 
+/**
+ * @brief Computes the top k values within each segment of a column
+ *
+ * This performs the equivalent of a sort and the slice of the resulting first k elements
+ * within each segment.
+ * However, the values within each segment may not necessarily be sorted.
```

Comment:
Should we note that a segmented sort can be used in post-processing to get a sorted answer?